src/controllers/controller.radar.js

```diff
@@ -85,43 +72,122 @@ module.exports = DatasetController.extend({
 		var custom = point.custom || {};
 		var dataset = me.getDataset();
 		var scale = me.chart.scale;
-		var pointElementOptions = me.chart.options.elements.point;
 		var pointPosition = scale.getPointPositionForValue(index, dataset.data[index]);
+		var x, y;
+
+		var options = me._resolvePointOptions(point, index);
+
+		x = reset ? scale.xCenter : pointPosition.x;
+		y = reset ? scale.yCenter : pointPosition.y;
+
+		// Utility
+		point._scale = scale;
+		point._options = options;
+		point._datasetIndex = me.index;
+		point._index = index;
+
+		// Desired view properties
+		point._model = {
+			x: x, // value not used in dataset scale, but we want a consistent API between scales
+			y: y,
+			skip: custom.skip || isNaN(x) || isNaN(y),
+			// Appearance
+			radius: options.radius,
+			pointStyle: options.pointStyle,
+			rotation: options.rotation,
+			backgroundColor: options.backgroundColor,
+			borderColor: options.borderColor,
+			borderWidth: options.borderWidth,
+			tension: resolve([custom.tension, dataset.lineTension, me.chart.options.elements.line.tension]),
+
+			// Tooltip
+			hitRadius: options.hitRadius
+		};
+	},
 
-		// Compatibility: If the properties are defined with only the old name, use those values
-		if ((dataset.radius !== undefined) && (dataset.pointRadius === undefined)) {
-			dataset.pointRadius = dataset.radius;
+	/**
+	 * @private
+	 */
+	_resolvePointOptions: function(element, index) {
```

Comment:
It looks to me like this is exactly the line controller's `_resolvePointOptions`: https://github.com/chartjs/Chart.js/blob/master/src/controllers/controller.line.js#L130

Can we share these methods somehow? Maybe we could make a `helpers._resolvePointOptions(chart, element, datasetIndex, dataIndex)`?

Reply:
I think correct place for the resolve... would be the element itself.

Reply:
But then the element needs a reference to the chart, which I think we should avoid. Doing it in the controller like we do now makes sense to me. But I'm wondering if we should move it to a base class or a helper

Reply:
Right, I was a bit hasty. 

I think correct place for option names that should be resolved is in the element, because only the element knows what to do with those options.
If a feature option is added to a element, it should be enough to add new option (and implementation of it) to element and configuration should be available in all places right away.

Correct place for the path to resolve is in controller. 
Context is also controller stuff.

I don't think this can be done in a good way in v2, so maybe those `_resolvePointOptions`, `_resolveLineOptions` etc should be placed in `DatasetController` instead.

I  any case, don't think we should list option names or the resolution path in a helper.

Reply:
One way to do this would be to have the radar controller inherit from the line controller but that may introduce too much coupling between them

Reply:
It seems natural to me to placing `_resolve*Options` in `DatasetController` and to have each controller define a different set of keys to resolve.

Reply:
That makes sense to me. We might not want a `_resolvePointOptions` in the `DatasetController` because many controllers will not have points (e.g. bar), but we could create a `_resolveElementOptions` and pass in the element type as a parameter along with the keys. And I like the idea of putting the keys / option names in `element.point.js` as @kurkle suggested

Reply:
What are we trying to refactor, the `for` loop? Maybe the context creation could be in a separate method (not even sure it's actually better), but since most controllers fallback differently, I don't think we need to refactor all these`_resolve*` methods, it could be worse actually.

About `_resolvePointOptions`, I think I agree with @etimberg, the radar and line controller share a lot of code and I would investigate refactoring these 2 controllers instead.

@nagix based on the previous comments, we may not be all agree on the way to share (or not) this code, so let's keep the refactor out of this PR and focus on the scriptable feature.

Reply:
Agreed

Reply:
> What are we trying to refactor, the `for` loop?

The line controller and radar controller share the same `_resolvePointOptions` character for character.

There are too many subtle differences between the controllers. I don't think it will be possible to merge them (at least in v2)

Reply:
> What are we trying to refactor, the for loop?

I was referring to your suggestion: "... but we could create a `_resolveElementOptions` and ..."